### ISSUE TITLE
Issue with making GetTransactionDetails for a Refund

### ISSUE THREAD

**Issue by tankchintan** [paypal/merchant-sdk-java#26]
I am trying to get details on an already completed refund by making a call to GetTransactionDetails API specified here https://developer.paypal.com/docs/classic/api/merchant/GetTransactionDetails_API_Operation_NVP/

But this is throwing exception when I use the Java SDK to do this. Weird thing is the logger is printing the refund details but the very next moment an exception is thrown.

Here is the XML which is printed right before the exception happens,
https://gist.github.com/tankchintan/d482b50e2d5cb5e0df79

Here is the exception,
```
SEVERE: java.lang.IllegalArgumentException: refund
at urn.ebay.apis.eBLBaseComponents.PaymentTransactionCodeType.fromValue(PaymentTransactionCodeType.java:63)
at urn.ebay.apis.eBLBaseComponents.PaymentInfoType.<init>(PaymentInfoType.java:1063)
at urn.ebay.apis.eBLBaseComponents.PaymentTransactionType.<init>(PaymentTransactionType.java:365)
at urn.ebay.api.PayPalAPI.GetTransactionDetailsResponseType.<init>(GetTransactionDetailsResponseType.java:96)
at urn.ebay.api.PayPalAPI.PayPalAPIInterfaceServiceService.getTransactionDetails(PayPalAPIInterfaceServiceService.java:595)
at urn.ebay.api.PayPalAPI.PayPalAPIInterfaceServiceService.getTransactionDetails(PayPalAPIInterfaceServiceService.java:542)
at com.snagfilms.api.external.payments.paypal.PayPalApiController.getRefundDetails(PayPalApiController.java:354)
at com.snagfilms.api.rest.controller.payments.UtilitiesController.getRefundDetails_aroundBody10(UtilitiesController.java:253)
at com.snagfilms.api.rest.controller.payments.UtilitiesController.getRefundDetails_aroundBody11$advice(UtilitiesController.java:60)
at com.snagfilms.api.rest.controller.payments.UtilitiesController.getRefundDetails_aroundBody12(UtilitiesController.java:1)
at com.snagfilms.api.rest.controller.payments.UtilitiesController.getRefundDetails_aroundBody13$advice(UtilitiesController.java:34)
at com.snagfilms.api.rest.controller.payments.UtilitiesController.getRefundDetails(UtilitiesController.java:1)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at org.springframework.web.method.support.InvocableHandlerMethod.invoke(InvocableHandlerMethod.java:213)
at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:126)
```

**Comment by juwlee** [comment]
@tankchintan thanks for reporting this issue. It seems the error is thrown when parsing the reponse. Can you please capture and post the response here?

**Comment by tankchintan** [comment]
```
<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns:cc="urn:ebay:apis:CoreComponentTypes" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" xmlns:ebl="urn:ebay:apis:eBLBaseComponents" xmlns:ed="urn:ebay:apis:EnhancedDataTypes" xmlns:ns="urn:ebay:api:PayPalAPI" xmlns:saml="urn:oasis:names:tc:SAML:1.0:assertion" xmlns:wsse="http://schemas.xmlsoap.org/ws/2002/12/secext" xmlns:wsu="http://schemas.xmlsoap.org/ws/2002/07/utility" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <SOAP-ENV:Header>
        <Security xmlns="http://schemas.xmlsoap.org/ws/2002/12/secext" xsi:type="wsse:SecurityType" />
        <RequesterCredentials xmlns="urn:ebay:api:PayPalAPI" xsi:type="ebl:CustomSecurityHeaderType">
            <Credentials xmlns="urn:ebay:apis:eBLBaseComponents" xsi:type="ebl:UserIdPasswordType">
                <Username xsi:type="xs:string" />
                <Password xsi:type="xs:string" />
                <Signature xsi:type="xs:string" />
                <Subject xsi:type="xs:string" />
            </Credentials>
        </RequesterCredentials>
    </SOAP-ENV:Header>
    <SOAP-ENV:Body id="_0">
        <GetTransactionDetailsResponse xmlns="urn:ebay:api:PayPalAPI">
            <Timestamp xmlns="urn:ebay:apis:eBLBaseComponents">2015-07-30T20:09:05Z</Timestamp>
            <Ack xmlns="urn:ebay:apis:eBLBaseComponents">Success</Ack>
            <CorrelationID xmlns="urn:ebay:apis:eBLBaseComponents">6447023dd87ae</CorrelationID>
            <Version xmlns="urn:ebay:apis:eBLBaseComponents">119.0</Version>
            <Build xmlns="urn:ebay:apis:eBLBaseComponents">000000</Build>
            <PaymentTransactionDetails xmlns="urn:ebay:apis:eBLBaseComponents" xsi:type="ebl:PaymentTransactionType">
                <ReceiverInfo xsi:type="ebl:ReceiverInfoType">
                    <Business xsi:type="ebl:EmailAddressType" />
                    <Receiver xsi:type="ebl:EmailAddressType">chintan+cf.pa.s@coolflix.com</Receiver>
                    <ReceiverID xsi:type="ebl:UserIDType">UDC6MWUNVWVNU</ReceiverID>
                </ReceiverInfo>
                <PayerInfo xsi:type="ebl:PayerInfoType">
                    <Payer xsi:type="ebl:EmailAddressType">chintan+cf.pa.s.per@coolflix.com</Payer>
                    <PayerID xsi:type="ebl:UserIDType">HSPXVLWP67JWW</PayerID>
                    <PayerStatus xsi:type="ebl:PayPalUserStatusCodeType">verified</PayerStatus>
                    <PayerName xsi:type="ebl:PersonNameType">
                        <Salutation />
                        <FirstName>Chintan CF</FirstName>
                        <MiddleName />
                        <LastName>PA SB Per</LastName>
                        <Suffix />
                    </PayerName>
                    <PayerCountry xsi:type="ebl:CountryCodeType">US</PayerCountry>
                    <PayerBusiness xsi:type="xs:string" />
                    <Address xsi:type="ebl:AddressType">
                        <Name xsi:type="xs:string" />
                        <Street1 xsi:type="xs:string" />
                        <Street2 xsi:type="xs:string" />
                        <CityName xsi:type="xs:string" />
                        <StateOrProvince xsi:type="xs:string" />
                        <CountryName />
                        <PostalCode xsi:type="xs:string" />
                        <AddressOwner xsi:type="ebl:AddressOwnerCodeType">PayPal</AddressOwner>
                        <AddressStatus xsi:type="ebl:AddressStatusCodeType">Unconfirmed</AddressStatus>
                    </Address>
                    <ContactPhone xsi:type="xs:string" />
                </PayerInfo>
                <TPLReferenceID xsi:type="xs:string" />
                <PaymentInfo xsi:type="ebl:PaymentInfoType">
                    <TransactionID>2YG71365EC105124F</TransactionID>
                    <ParentTransactionID xsi:type="ebl:TransactionId">2AR9296116674132T</ParentTransactionID>
                    <ReceiptID />
                    <TransactionType xsi:type="ebl:PaymentTransactionCodeType">refund</TransactionType>
                    <PaymentType xsi:type="ebl:PaymentCodeType">instant</PaymentType>
                    <PaymentDate xsi:type="xs:dateTime">2015-04-27T22:44:42Z</PaymentDate>
                    <GrossAmount xsi:type="cc:BasicAmountType" currencyID="USD">-1.00</GrossAmount>
                    <FeeAmount xsi:type="cc:BasicAmountType" currencyID="USD">0.05</FeeAmount>
                    <TaxAmount xsi:type="cc:BasicAmountType" currencyID="USD">0.82</TaxAmount>
                    <ExchangeRate xsi:type="xs:string" />
                    <PaymentStatus xsi:type="ebl:PaymentStatusCodeType">Completed</PaymentStatus>
                    <PendingReason xsi:type="ebl:PendingStatusCodeType">none</PendingReason>
                    <ReasonCode xsi:type="ebl:ReversalReasonCodeType">none</ReasonCode>
                    <ProtectionEligibility xsi:type="xs:string">Ineligible</ProtectionEligibility>
                    <ProtectionEligibilityType xsi:type="xs:string">None</ProtectionEligibilityType>
                    <ShipAmount xsi:type="xs:string" />
                    <ShipHandleAmount xsi:type="xs:string" />
                    <ShipDiscount xsi:type="xs:string" />
                    <InsuranceAmount xsi:type="xs:string" />
                    <Subject xsi:type="xs:string" />
                    <ReceiptReferenceNumber xsi:type="xs:string" />
                    <POSTransactionType xsi:type="xs:string" />
                    <StoreID xsi:type="xs:string" />
                    <TerminalID xsi:type="xs:string" />
                    <BinEligibility xsi:type="xs:string" />
                </PaymentInfo>
                <PaymentItemInfo xsi:type="ebl:PaymentItemInfoType">
                    <InvoiceID xsi:type="xs:string">CIN-RE-PAWE-0000014C-1430174679</InvoiceID>
                    <Custom xsi:type="xs:string">CIN-CA-PAWE-0000014C-1429920693</Custom>
                    <Memo xsi:type="xs:string">testing paypal refund 2</Memo>
                    <SalesTax xsi:type="xs:string">0.82</SalesTax>
                    <PaymentItem xsi:type="ebl:PaymentItemType">
                        <EbayItemTxnId xsi:type="xs:string" />
                        <Name xsi:type="xs:string">DUMMY! CineRoar more expensive then 10 plan: null</Name>
                        <Number xsi:type="xs:string" />
                        <Quantity xsi:type="xs:string">1</Quantity>
                        <SalesTax xsi:type="xs:string">0.82</SalesTax>
                        <ShippingAmount xsi:type="xs:string" />
                        <HandlingAmount xsi:type="xs:string" />
                        <CouponID xsi:type="xs:string" />
                        <CouponAmount xsi:type="xs:string" />
                        <CouponAmountCurrency xsi:type="xs:string" />
                        <LoyaltyCardDiscountAmount xsi:type="xs:string" />
                        <LoyaltyCardDiscountCurrency xsi:type="xs:string" />
                        <InvoiceItem xsi:type="ebl:InvoiceItemType">
                            <ItemCountUnit xsi:type="ebl:UnitOfMeasure" />
                            <Taxable xsi:type="xs:string">false</Taxable>
                            <TaxRate xsi:type="xs:string" />
                            <MPN xsi:type="xs:string" />
                            <ISBN xsi:type="xs:string" />
                            <PLU xsi:type="xs:string" />
                            <ModelNumber xsi:type="xs:string" />
                            <StyleNumber xsi:type="xs:string" />
                        </InvoiceItem>
                    </PaymentItem>
                    <Subscription xsi:type="ebl:SubscriptionInfoType" recurring="" reattempt="">
                        <SubscriptionID />
                        <Username xsi:type="xs:string" />
                        <Password xsi:type="xs:string" />
                        <Recurrences xsi:type="xs:string" />
                    </Subscription>
                    <Auction xsi:type="ebl:AuctionInfoType" multiItem="">
                        <BuyerID xsi:type="xs:string" />
                    </Auction>
                </PaymentItemInfo>
                <OfferCouponInfo xsi:type="ebl:OfferCouponInfoType">
                    <Type xsi:type="xs:string" />
                    <ID xsi:type="xs:string" />
                    <Amount xsi:type="xs:string" />
                    <AmountCurrency xsi:type="xs:string" />
                </OfferCouponInfo>
            </PaymentTransactionDetails>
        </GetTransactionDetailsResponse>
    </SOAP-ENV:Body>
</SOAP-ENV:Envelope>
```

**Comment by tankchintan** [comment]
Hi @juwlee any updates on this issue, if it's going to be resolved anytime soon?

**Comment by juwlee** [comment]
@tankchintan we are looking into it with the API team under an internal ticket(SRE-32785). We will update this thread with any news.

**Comment by juwlee** [comment]
Fixed in 04f22b2db177b5cd54f7ef39251c70161c4fb889
